src/js/ripple/orderbook.js

```diff
@@ -1176,4 +1339,347 @@ OrderBook.prototype.is_valid = function() {
+    this.setLegOneTakerGetsFunded(legOneOffer, legOneOffer.initTakerGetsFunded);
+  } else {
+    var updatedLeftover = legOneOffer.initTakerGetsFunded.subtract(takerGets);
+
+    this.setLegOneTakerGetsFunded(legOneOffer, takerGets);
+    this.addLeftoverOwnerFunds(legOneOffer.Account, updatedLeftover);
+  }
+};
+
+/**
+ * Increase leg one offer funded amount with extra funds found after applying
+ * clamp.
+ *
+ * @param [Object] legOneOffer - IOU:XRP offer
+ */
+
+OrderBook.prototype.adjustLegOneFundedAmount = function(legOneOffer) {
+  assertValidLegOneOffer(legOneOffer, 'Leg one offer is invalid');
+  assert(!legOneOffer.is_fully_funded, 'Leg one offer cannot be fully funded');
+
+  var fundedSum = this.getOfferTakerGetsFunded(legOneOffer)
+    .add(this.getLeftoverOwnerFunds(legOneOffer.Account));
+
+  if (fundedSum.compareTo(this.getOfferTakerGets(legOneOffer)) >= 0) {
+    // There are enough extra funds to fully fund the offer
+    var legOneTakerGets = this.getOfferTakerGets(legOneOffer);
+    var updatedLeftover = fundedSum.subtract(legOneTakerGets);
+
+    this.setLegOneTakerGetsFunded(legOneOffer, legOneTakerGets);
+    this.setLeftoverOwnerFunds(legOneOffer.Account, updatedLeftover);
+  } else {
+    // There are not enough extra funds to fully fund the offer
+    this.setLegOneTakerGetsFunded(legOneOffer, fundedSum);
+    this.resetOwnerFundsLeftover(legOneOffer.Account);
+  }
+};
+
+/**
+ * Set taker gets funded amount for a IOU:XRP offer. Also calculates taker
+ * pays funded using offer quality and updates is_fully_funded flag
+ *
+ * @param [Object] legOneOffer - IOU:XRP offer
+ * @param [Amount] takerGetsFunded
+ */
+
+OrderBook.prototype.setLegOneTakerGetsFunded = setLegOneTakerGetsFunded;
+
+function setLegOneTakerGetsFunded(legOneOffer, takerGetsFunded) {
+  assertValidLegOneOffer(legOneOffer, 'Leg one offer is invalid');
+  assert(takerGetsFunded instanceof Amount, 'Taker gets funded is invalid');
+
+  legOneOffer.taker_gets_funded = takerGetsFunded.to_text();
+  legOneOffer.taker_pays_funded = takerGetsFunded
+    .multiply(this.getOfferQuality(legOneOffer))
+    .to_text();
+
+  if (legOneOffer.taker_gets_funded === legOneOffer.TakerGets.value) {
+    legOneOffer.is_fully_funded = true;
+  }
+}
+
+/**
+ * Set taker gets amount for a IOU:XRP offer. Also calculates taker pays
+ * using offer quality
+ *
+ * @param [Object] legOneOffer - IOU:XRP offer
+ * @param [Amount] takerGets
+ */
+
+OrderBook.prototype.setLegOneTakerGets = function(legOneOffer, takerGets) {
+  assertValidLegOneOffer(legOneOffer, 'Leg one offer is invalid');
+  assert(takerGets instanceof Amount, 'Taker gets funded is invalid');
+
+  legOneOffer.TakerGets = takerGets.to_text();
+  legOneOffer.TakerPays = takerGets.multiply(this.getOfferQuality(legOneOffer));
+};
+
+/**
+ * Merge direct and autobridged offers into a combined orderbook
+ *
+ * @return [Array]
+ */
+
+OrderBook.prototype.mergeDirectAndAutobridgedBooks = function() {
```

Comment:
I think most of this function can be replaced with
`this._mergedOffers = _.sortBy(this._offers.concat(this._offersAutobridged), this.getOfferQuality);`